File: server/src/test/java/org/magpie/mt/util/SegmentStringTest.java
package org.magpie.mt.util;

import org.junit.Test;
import org.magpie.mt.api.dto.LocaleCode;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Alex Eng<a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
 */
public class SegmentStringTest {

    @Test
    public void segmentStringDefaultLocaleTest() {
        String text =
                "A cooled coincidence parks the spectacular. Behind a dragon suspects the roman fashion. How can the inhabited wallet provision her made journalist? The opera squashes the younger advantage.";
        List<String> strings = SegmentString.segmentString(text, Optional.empty());
        assertThat(strings.size()).isEqualTo(4);
    }

    @Test
    public void segmentStringTest() {
        String text =
                "Das nennen Sie ein Schinken-Sandwich? Nehmen Sie das sofort weg!";
        List<String> strings = SegmentString.segmentString(text, Optional.of(
                LocaleCode.DE));
        assertThat(strings.size()).isEqualTo(2);
    }
}

File: server/src/test/java/org/zanata/magpie/util/SegmentStringTest.kt
package org.zanata.magpie.util

import org.junit.Test
import org.zanata.magpie.api.dto.LocaleCode
import java.util.Optional

import org.assertj.core.api.Assertions.assertThat

/**
 * @author Alex Eng[aeng@redhat.com](mailto:aeng@redhat.com)
 */
class SegmentStringTest {

    @Test
    fun segmentStringDefaultLocaleTest() {
        val text = """
            |A cooled coincidence parks the spectacular. Behind a dragon suspects the roman fashion.
            |How can the inhabited wallet provision her made journalist?
            |The opera squashes the younger advantage.""".trimMargin()
        val strings = segmentBySentences(text, Optional.empty())
        assertThat(strings).containsExactly(
                "A cooled coincidence parks the spectacular. ",
                "Behind a dragon suspects the roman fashion.\n",
                "How can the inhabited wallet provision her made journalist?\n",
                "The opera squashes the younger advantage.")
    }

    @Test
    fun segmentStringTest() {
        val text = "Das nennen Sie ein Schinken-Sandwich? Nehmen Sie das sofort weg!"
        val strings = segmentBySentences(text, Optional.of(LocaleCode.DE))
        assertThat(strings).containsExactly(
                "Das nennen Sie ein Schinken-Sandwich? ",
                "Nehmen Sie das sofort weg!")
    }

    @Test
    fun segmentTextWithInlineHTML() {
        val text = "Jaguar will sell its new XJ-6 model in the U.S. for a small fortune :-). \n" +
                "Expect to pay around USD 120ks. Custom options can set you back another \n" +
                "few 10,000 dollars. For details, go to <a href=\"http://www.jaguar.com/sales\" \n" +
                "alt=\"Click here\">Jaguar Sales</a> or contact xj-6@jaguar.com."
        val strings = segmentBySentences(text, Optional.of(LocaleCode.EN))
//        for ((i, s) in strings.withIndex()) { println("$i: $s") }

        assertThat(strings).containsExactly(
                "Jaguar will sell its new XJ-6 model in the U.S. for a small fortune :-). \n",
                "Expect to pay around USD 120ks. ",
                "Custom options can set you back another \n" +
                "few 10,000 dollars. ",
                "For details, go to <a href=\"http://www.jaguar.com/sales\" \n" +
                "alt=\"Click here\">Jaguar Sales</a> or contact xj-6@jaguar.com.")
    }
}
